### PR TITLE
Feature: Join party and deep-linking 🔥

### LINKED ISSUE #20: Join party
Join the party, check #18 to make sure of the data needed.

- [x] Join using the party PIN.
- [x] Join using the party QR.

### LINKED ISSUE #30: Register deep link
Register the deep link used in the qr code.

### PR DESCRIPTION
**Pull request by wovalle**
Adding Join party and Deep linking.

Closes #20 
Closes #30

### CODE REVIEW

**Comment by vetom** [comment]
Don't merge into the branch `feature/create-party`, wait for the review and close of #29 and then merge this changes directly to `master`.

File: src/views/Join/index.js [hunk 1]
```diff
@@ -1,19 +1,67 @@
...
+    const { partyId } = this.state;
+    const party = await this.db.getPartyById(partyId);
+
+    console.log({ party });
+
+    if (!party) {
+      alert(`No party with id ${partyId}`);
+      return;
+    }
+
```
Comment (vetom) [inline src/views/Join/index.js]: Here is a great place to handle #32 for players.

File: src/views/Join/index.js [hunk 2]
```diff
@@ -1,30 +1,158 @@
...
             bordered
             info
             style={styles.buttonBlock}
-            onPress={() => this.props.navigation.navigate('Lobby')}
+            onPress={this.openCamera}
           >
-            <Text> Scan </Text>
-            <Icon name='md-qr-scanner' />
+            <Text>Scan</Text>
+            <Icon name="md-qr-scanner" />
```
Comment (vetom) [inline src/views/Join/index.js]: Should we update our prettier config file to use single quotes?

Comment (Randyjp) [reply src/views/Join/index.js]: Airbnb style-guide uses single quotes for strings but double quotes for HTML/JSX attributes.  I think it's nice to have that small difference but we can change it.

**Pull request merged**